colossalai/fx/profiler/profiler.py

```diff
@@ -1,15 +1,21 @@
+from functools import partial
 from typing import Callable, Any, Dict, Tuple
 import torch
 from torch.fx import Graph, Node
 from torch.fx.node import Argument, Target
 from torch.utils._pytree import tree_map
-from .dataflow import GraphInfo, autograd_graph_analysis, Phase
-from .memory import WEIRD_OPS
+from .dataflow import autograd_graph_analysis, is_phase, Phase, GraphInfo
+from .memory import activation_size
+from .constant import ALIAS_ATEN
 from .tensor import MetaTensor
 from .opcount import flop_mapping
 
 __all__ = ['profile_function', 'profile_module', 'profile_method']
 
+# super-dainiu: this cache should be global, otherwise it cannot
+# track duplicated tensors between nodes
+cache = set()
```

Comment:
this cache should be global, otherwise it cannot track duplicated tensors between nodes